### PR TITLE
[#158180318] Update the chrome plugin

### PR DESCRIPTION
**Pull request by paroxp**
What
----

Now that we have upgraded concourse, some of the frontend has changed.
This means our plugin is out of date and should be updated if we'd like
to keep on using it on our dashboard TV.

In the new concourse version, upstream has decided to separate the
grouping bar into it's own place. This is a simple change really and
requires some minimal effort to hide new additional element.

Some other changes have been made, purely as a maintenance task to
keep the linter happy as well as the code more modern.

See commit for more details.

How to review
-------------

- Code review
- Copy contents of the file
- Open the deployer concourse on staging
- Open the `Inspect` console on Chrome (Cmd + Alt + J)
- Paste the contents of the file
- See no errors and expect the dashboard to be clean

After merge
-----------

Make sure to update the plugin on the TV's Chrome. See README for more details.

### CODE REVIEW

**Comment by alext** [comment]
This change seems to conflate 2 things - there are some JS style changes, and the functional changes to reflect the newer concourse. Could they be separated into 2 commits so that we can more easily see what the functional changes are.

**Comment by paroxp** [comment]
I was hoping this would go unseen 👀 

Sure!

File: misc/chrome_plugins/clean_concourse_pipeline/src/inject.user.js [hunk 1]
```diff
@@ -8,27 +8,27 @@
...
+const readyStateCheckInterval = setInterval(() => {
+  if (document.readyState === 'complete') {
+    clearInterval(readyStateCheckInterval);
 
-        console.log("Monitor mode is go");
-        var element = document.getElementsByClassName("legend")[0];
-        element.parentNode.removeChild(element);
+    console.log('Monitor mode is go');
+    const $legent = document.querySelector('.legend');
+    $legent.parentNode.removeChild($legent);
```
Comment (richardTowers) [inline misc/chrome_plugins/clean_concourse_pipeline/src/inject.user.js]: `$legent` typo?

Comment (paroxp) [reply misc/chrome_plugins/clean_concourse_pipeline/src/inject.user.js]: thanks

**Comment by richardTowers** [comment]
The script looks like it works (it removes the chrome as you'd expect), but after it's run I see a bunch of errors from the elm:

```
elm.min.js?id=18701b041d48f77c150ba19dedd849da:1 Uncaught TypeError: Cannot read property 'childNodes' of undefined
```

Presumably we're removing some elements that concourse still thinks it owns. Would it be better to set `display: none;` on things instead of removing them?

Do we care about these errors? I wouldn't have raised them, but you said "See no errors" in your instructions :)

**Comment by paroxp** [comment]
> I wouldn't have raised them, but you said "See no errors" in your instructions

You're right, my bad :D

In general, I didn't really care for it. But it does mean the console is cleaner and would make it easier to debug things.

**Pull request merged**